tests/test_core.py

```diff
@@ -4824,9 +4824,9 @@ def test():
       print('(iteration)')
       time.sleep(random.random()/(10*num)) # add some timing nondeterminism here, not that we need it, but whatever
       self.do_run(src, 'hello world\n77.\n')
-      ret = open('src.cpp.o.js').read()
+      ret = open('src.cpp.o.js', 'rb').read()
```

Comment:
this one is a text file, though?

Reply:
Yes, but `open()` without `rb` generates Unicode string on Python 3 and thus causes error when `ret += (binary byte string)`.

Reply:
We are not actually accessing the inner texts so I think we can safely read it as a binary file.